vsphere/datadog_checks/vsphere/legacy/vsphere_legacy.py

```diff
@@ -394,68 +393,58 @@ def _get_parent_tags(self, mor, all_objects):
             return parent_tags
         return []
 
-    @staticmethod
-    @contextmanager
-    def create_container_view(server_instance, resources):
-        content = server_instance.content
-        view_ref = content.viewManager.CreateContainerView(content.rootFolder, resources, True)
-        try:
-            yield view_ref
-        finally:
-            view_ref.Destroy()
-
     def _collect_mors_and_attributes(self, server_instance):
         resources = list(RESOURCE_TYPE_METRICS)
         resources.extend(RESOURCE_TYPE_NO_METRIC)
+
         content = server_instance.content
+        view_ref = content.viewManager.CreateContainerView(content.rootFolder, resources, True)
 
-        with VSphereCheck.create_container_view(server_instance, resources) as view_ref:
```

Comment:
Can we move the changes of the legacy implementation to another separate PR?

In case, they is an issue related to this change. It will be easier to investigate.

Reply:
Back to minimal changes